manager/integration/tests/test_scheduling.py

```diff
@@ -881,6 +880,7 @@ def test_data_locality_basic(client, core_api, volume_name, pod, settings_reset)
     v4_node3_replica_count = 0
 
     for replica in volume4.replicas:
+        wait_for_replica_running(client, volume4_name, replica["name"])
```

Comment:
Additional wait to check replica is in running state is introduced before checking the ```hostId``` of the replica because immediately after ```wait_for_volume_replica_count``` method completes, we don't have the ```hostId``` and other details available in the replica.
```
[{'address': '', 'currentImage': '', 'dataPath': '', 'diskID': '', 'engineImage': 'longhornio/longhorn-engine:master', 'failedAt': '', 'hostId': '', 'instanceManagerName': '', 'mode': '', 'name': 'longhorn-testvol-l2gpw4-4-r-3caf1504', 'running': False}, {'address': 'xx.xx.xx.xx', 'currentImage': 'longhornio/longhorn-engine:master', 'dataPath': '/var/lib/longhorn/replicas/longhorn-testvol-l2gpw4-4-a776fd90', 'diskID': 'default-disk-3e72533a17f25ad4', 'engineImage': 'longhornio/longhorn-engine:master', 'failedAt': '', 'hostId': 'node-wk2', 'instanceManagerName': 'instance-manager-r-3cc712b8', 'mode': 'RW', 'name': 'longhorn-testvol-l2gpw4-4-r-c67fb79d', 'running': True}]
```

Reply:
Does that mean we need to wait for replica running for line 765 as well?

Reply:
This is a case where the replica count has just increased and ```wait_for_volume_replica_count``` returns the correct count but the ```hostId``` and other value are yet to be assigned to the replica. 
Whereas at line 765 the replica is being deleted and once  ```wait_for_volume_replica_count``` confirms that we'll not need ```wait_for_replica_running```.